lib/integrations/awsoidc/eks_enroll_clusters.go

```diff
@@ -0,0 +1,528 @@
+			CAData: eksClusterCA,
+		},
+	}
+
+	configFlags := genericclioptions.NewConfigFlags(false)
+	configFlags.WithWrapConfigFn(func(*rest.Config) *rest.Config {
+		return restConfig
+	})
+
+	return configFlags, nil
+}
+
+func getHelmActionConfig(clientGetter genericclioptions.RESTClientGetter, log logrus.FieldLogger) (*action.Configuration, error) {
+	actionConfig := new(action.Configuration)
+	if err := actionConfig.Init(clientGetter, agentNamespace, "secret", log.WithField("source", "helm").Debugf); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return actionConfig, nil
+}
+
+func getHelmSettings() *helmCli.EnvSettings {
+	helmSettings := helmCli.New()
+	helmSettings.RepositoryCache = os.TempDir()
+	helmSettings.SetNamespace(agentNamespace)
+
+	return helmSettings
+}
+
+// checkAgentAlreadyInstalled checks through the Helm if teleport-kube-agent chart was already installed in the EKS cluster.
+func checkAgentAlreadyInstalled(actionConfig *action.Configuration) (bool, error) {
+	listCmd := action.NewList(actionConfig)
+
+	releases, err := listCmd.Run()
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+	for _, release := range releases {
+		if release.Name == agentName {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// installKubeAgent installs teleport-kube-agent chart to the target EKS cluster.
+func installKubeAgent(ctx context.Context, eksCluster *eksTypes.Cluster, proxyAddr, joinToken, resourceId string, actionConfig *action.Configuration, settings *helmCli.EnvSettings, req EnrollEKSClustersRequest) error {
+	installCmd := action.NewInstall(actionConfig)
+	installCmd.RepoURL = agentRepoURL
+	installCmd.Version = req.AgentVersion
+	if strings.Contains(installCmd.Version, "dev") {
+		installCmd.Version = "" // For testing during development.
+	}
+
+	chartPath, err := installCmd.LocateChart(agentName, settings)
+	if err != nil {
+		return trace.Wrap(err, "could not locate chart")
+	}
+	defer func() {
+		// Clean up downloaded chart.
+		_ = os.Remove(chartPath)
+	}()
+
+	agentChart, err := loader.Load(chartPath)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	installCmd.ReleaseName = agentName
+	installCmd.Namespace = agentNamespace
+	installCmd.CreateNamespace = true
+	vals := map[string]any{}
+	vals["proxyAddr"] = proxyAddr
+
+	vals["roles"] = "kube"
+	// todo(anton): Remove check for 13 once Teleport cloud is unblocked to move from v13 chart.
+	if req.EnableAppDiscovery && !strings.HasPrefix(installCmd.Version, "13") {
+		vals["roles"] = "kube,app,discovery"
+	}
+	vals["authToken"] = joinToken
+
+	if req.IsCloud && req.EnableAutoUpgrades {
+		vals["updater"] = map[string]any{"enabled": true, "releaseChannel": "stable/cloud"}
+
+		vals["highAvailability"] = map[string]any{"replicaCount": 2,
+			"podDisruptionBudget": map[string]any{"enabled": true, "minAvailable": 1},
+		}
+	}
+
+	eksTags := make(map[string]*string, len(eksCluster.Tags))
+	for k, v := range eksCluster.Tags {
+		eksTags[k] = &v
```

Comment:
```suggestion
		eksTags[k] = aws.String(v)
```

This will make all keys use the same exact value (the last one)
https://go.dev/play/p/zZ2LC2cruXM

Reply:
I'm just planning on it running on go1.22... 😎  (it got me 😭 )